exporter/otlpexporter/otlp.go

```diff
@@ -102,14 +108,17 @@ func (e *baseExporter) pushTraces(ctx context.Context, td ptrace.Traces) error {
 			zap.Int64("dropped_spans", resp.PartialSuccess().RejectedSpans()),
 		)
 	}
-	return nil
+	return
 }
 
-func (e *baseExporter) pushMetrics(ctx context.Context, md pmetric.Metrics) error {
+func (e *baseExporter) pushMetrics(ctx context.Context, md pmetric.Metrics) (err error) {
+	defer func() {
+		e.reportStatusFromError(err)
+	}()
```

Comment:
I would do this way, because it is much easier to read:

```suggestion
func (e *baseExporter) pushMetricsWithStatus(ctx context.Context, md pmetric.Metrics) err {
	err := e.pushMetrics(ctx, md)
	if err != nil {
		componentstatus.ReportStatus(e.host, componentstatus.NewRecoverableErrorEvent(err))
	}
	return err
}

func (e *baseExporter) pushMetrics(ctx context.Context, md pmetric.Metrics) error {
```